src/context-providers/builder/provider.tsx

```diff
@@ -48,8 +52,10 @@ export const BuilderProvider = ({
     const [state, dispatch] = useImmerReducer(builderReducer, DEFAULT_VALUES);
 
     return (
-        <BuilderContext.Provider value={{ state, dispatch }}>
-            {children}
-        </BuilderContext.Provider>
+        <MainWrapper>
+            <BuilderContext.Provider value={{ state, dispatch }}>
+                {children}
+            </BuilderContext.Provider>
+        </MainWrapper>
```

Comment:
provider file should only render the necessary components for this context provider and not render ui elements, move `MainWrapper` out of this component